@utahdts/utah-design-system/react/components/buttons/ConfirmationButton.jsx

```diff
@@ -58,29 +87,95 @@ function ConfirmationButton({
   promptChildren,
   size,
   type,
+  ...rest
 }) {
   const [isClicked, setIsClicked] = useState(false);
-  const buttonClick = () => {
-    setIsClicked(!isClicked);
-    onClick();
-  };
+  const timer = useRef(NaN);
+  const { addAssertiveMessage } = useAriaMessaging();
+
+  const clearTimer = useCallback(() => {
+    if (timer.current) {
+      clearTimeout(timer.current);
+      timer.current = NaN;
+    }
+  }, []);
+
+  const resetButton = useCallback(() => {
```

Comment:
We need to reset the button when the user clicks off the button.

Reply:
Also, should probably have a `useEffect(() => () => clearTimeout(timer.current), [])` so as not to get errors when timeout expires if component is unmounted.

Reply:
> Also, should probably have a `useEffect(() => () => clearTimeout(timer.current), [])` so as not to get errors when timeout expires if component is unmounted.

Timer is dead, RIP.